crates/viewer/re_renderer/src/resource_managers/texture_manager.rs

```diff
@@ -201,54 +145,74 @@ impl Inner {
-            &device,
-            &queue,
-            texture_pool,
-            &Texture2DCreationDesc {
+        // Create the single pixel white texture ad hoc - at this point during initialization we don't have
+        // the render context yet and thus can't use the higher level `transfer_image_data_to_texture` function.
+        let white_texture_unorm = GpuTexture2D(texture_pool.alloc(
+            device,
+            &TextureDesc {
                 label: "white pixel - unorm".into(),
-                data: vec![255, 255, 255, 255].into(),
                 format: wgpu::TextureFormat::Rgba8Unorm,
+                size: wgpu::Extent3d {
+                    width: 1,
+                    height: 1,
+                    depth_or_array_layers: 1,
+                },
+                mip_level_count: 1,
+                sample_count: 1,
+                dimension: wgpu::TextureDimension::D2,
+                usage: wgpu::TextureUsages::TEXTURE_BINDING | wgpu::TextureUsages::COPY_DST,
+            },
+        ));
+        queue.write_texture(
+            wgpu::ImageCopyTexture {
+                texture: &white_texture_unorm.texture,
+                mip_level: 0,
+                origin: wgpu::Origin3d::ZERO,
+                aspect: wgpu::TextureAspect::All,
+            },
+            &[255, 255, 255, 255],
+            wgpu::ImageDataLayout {
+                offset: 0,
+                bytes_per_row: Some(4),
+                rows_per_image: None,
+            },
+            wgpu::Extent3d {
                 width: 1,
                 height: 1,
+                depth_or_array_layers: 1,
             },
-        )
-        .expect("Failed to create white pixel texture!");
+        );
 
         let zeroed_texture_float =
-            create_zero_texture(texture_pool, &device, wgpu::TextureFormat::Rgba8Unorm);
-        let zeroed_texture_depth =
-            create_zero_texture(texture_pool, &device, wgpu::TextureFormat::Depth16Unorm);
+            create_zero_texture(texture_pool, device, wgpu::TextureFormat::Rgba8Unorm);
         let zeroed_texture_sint =
-            create_zero_texture(texture_pool, &device, wgpu::TextureFormat::Rgba8Sint);
+            create_zero_texture(texture_pool, device, wgpu::TextureFormat::Rgba8Sint);
         let zeroed_texture_uint =
-            create_zero_texture(texture_pool, &device, wgpu::TextureFormat::Rgba8Uint);
+            create_zero_texture(texture_pool, device, wgpu::TextureFormat::Rgba8Uint);
 
         Self {
             white_texture_unorm,
             zeroed_texture_float,
-            zeroed_texture_depth,
             zeroed_texture_sint,
             zeroed_texture_uint,
-            device,
-            queue,
             inner: Default::default(),
         }
     }
 
     /// Creates a new 2D texture resource and schedules data upload to the GPU.
     /// TODO(jleibs): All usages of this should be replaced with `get_or_create`, which is strictly preferable
```

Comment:
and this one is surprisingly hard to get rid of because for that we'd to be sure that we get a unique enough description of the mesh without having to resort to a hash of the texture content within it.